website/source/guides/plugin-backends.html.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: "guides"
+page_title: "Plugin Backends - Guides"
+sidebar_current: "guides-plugin-backends"
+description: |-
+  Learn how to build, register, and mount a custom plugin backend.
+---
+
+# Introduction
+
+Plugin backends utilize the [plugin system][plugin-system] to enable 
+third-party secret and auth backends to be mounted. 
+
+It is worth noting that even though [database backend][database-backend]
+operates under the same underlying plugin mechanism, it is slightly different
+in design than plugin backends demonstrated in this guide. The database backend 
+manages multiple plugins under the same backend mount point, whereas plugin
+backends are generic backends that functions as either a secret or auth backend. 
+
+This guide provides steps to build, register, and mount non-database external
+plugin backends.
+
```

Comment:
Somewhere we need to mention that reading the original connection's TLS connection state is not supported in plugins.